griptape/tools/google_gmail_client/tool.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+import logging
+import base64
+from email.message import EmailMessage
+from schema import Schema, Literal, Optional
+from attr import define, field
+from griptape.artifacts import InfoArtifact, ErrorArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+
+
+@define
+class GoogleGmailClient(BaseGoogleClient):
+    CREATE_DRAFT_EMAIL_SCOPES = ['https://www.googleapis.com/auth/gmail.compose']
+
+    owner_email: str = field(kw_only=True)
+
+    @activity(config={
+        "description": "Can be used to create a draft email in GMail",
+        "schema": Schema({
+            Literal(
+                "to",
+                description="email address which to send to"
+            ): str,
+            Literal(
+                "subject",
+                description="subject of the email"
+            ): str,
+            Literal(
+                "body",
+                description="body of the email"
+            ): str,
+            Optional(
+                "attachment_names",
+                description="Names of the attachments"
+            ): list[str],
```

Comment:
I don't think this works. Needs to be a list of attachment objects with the following properties:
- attachment_name
- memory_name
- artifact_namespace

Reply:
It works actually